umpf

```diff
@@ -1042,7 +1044,10 @@ rebase_end() {
 		                    ) \)
 		                | \.                             # struct members etc.
 		                )*                               # function name + arguments etc.
-		            \s*),\s*
+		            \s*)
+		           | (?&expr_noternary) \? (?&expr_noternary) : (?&expr)
```

Comment:
Hmm, `\s*` instead of a single spaces? And why `expr` for the last argument and `expr_noternary` for the others?

Reply:
I'm not introducing `\s*`, just moving them around. They are there to support newlines and indentation.

The condition argument has to be  `expr_noternary`  to avoid infinite recursion.

The last `expr` is there to support nesting, e.g.:
```
a ? b : 
c ? d : e
```
This is not required by Mesa though.

Reply:
I meant the spaces within the expression. This won't match `a?b:c` right now (without spaces), or am I missing something?

Reply:
It does match `a?b:c`. The `x` modifier makes Perl ignore decorative whitespace in the regular expression. Whitespace is matched via `\s*` at the start and end of `expr_noternary` and `expr`.

I just noticed that Mesa `main` already [got rid of the ternary operator](https://gitlab.freedesktop.org/mesa/mesa/-/merge_requests/20752/diffs#0cc1139e3347f573ae1feee5b73dbc8a8a21fcfa_29_24) again. Now we are missing support for `files('VERSION')`.

Reply:
Right, `project` accepts `file` objects for `version` and we cannot concatenate those with strings.
I've experimented with it a bit and the only thing that I came up with that worked was:
```
version : run_command('cat', 'VERSION', check: true).stdout().strip() + '.20230208-1'
```

Not very nice. Other ideas?

But that's something for the future. This patch is fine, so I'll merge it.

Reply:
> Not very nice. Other ideas?

Parse the filename from `version : files('VERSION')`, then open that file and append the version in there?